src/lib/withGestures.jsx

```diff
@@ -5,19 +5,44 @@ import Hammer from 'hammerjs'
 const shouldListenToSwipe = handlers => Object.keys(handlers).find(h => /^swipe/.test(h)) !== undefined
 const shouldListenToTap = handlers => Object.keys(handlers).find(h => /^tap/.test(h)) !== undefined
 
+const GHOST_CLICK_DELAY = 350
+
 const withGestures = (eventHandlers) => {
   return WrappedComponent => {
     return class WithGesturesComponent extends Component {
       componentDidMount () {
-        this.hammer = new Hammer.Manager(ReactDOM.findDOMNode(this), {
+        let node = ReactDOM.findDOMNode(this);
+        this.hammer = new Hammer.Manager(node, {
           recognizers: [[Hammer.Tap], [Hammer.Swipe, { direction: Hammer.DIRECTION_ALL }]]
         })
         this.handlers = eventHandlers(this.props)
         if (shouldListenToSwipe(this.handlers)) {
           this.hammer.on('swipe', e => this.onSwipe(e))
         }
         if (shouldListenToTap(this.handlers)) {
-          this.hammer.on('tap', e => this.onTap(e))
+          // when listening for taps, we need to debounce the ghost click that happens right afterwards
+          let lastTouchEvent = null
+          let ghostClickTimeout = null
+          this.hammer.on('tap', e => {
+            lastTouchEvent = e
+            // if nothing happens in the next couple of ms, we fire the tap event
+            ghostClickTimeout = setTimeout(() => {
+              this.onTap(e)
+              ghostClickTimeout = lastTouchEvent = null
+            }, GHOST_CLICK_DELAY)
+          })
+          node.addEventListener('click', e => {
+            // when a click event happens, we check to see if it's related to the last touch event we received
+            if (ghostClickTimeout !== null &&
```

Comment:
Pure personal opinion, when a condition becomes pretty complex I like to store it in an explicit boolean variable like : 
```javascript
const eventAreRelated = ghostClickTimeout !== null && lastTouchEvent !==
  Math.round(lastTouchEvent.srcEvent.pageX) === Math.round(e.pageX) &&
  Math.round(lastTouchEvent.srcEvent.pageY) === Math.round(e.pageY)

if (eventAreRelated) {
  [...]
}
```
It allows us to not comment the code, and make it easier to read (I generally try to read only the left part of an affectation)